airootfs/usr/share/vx-img/flash-image.sh

```diff
@@ -78,31 +81,75 @@ for f in "$_path"/*; do
     fi
 done
 
+if [[ -z $_toflash ]]; then
+    echo "Found no image to flash. Exiting..."
+    exit
+fi
+
 echo "Extracting and flashing $_toflash"
 
-echo "What is the expected final size of the image? [64g]:"
+clear
+echo "What is the expected final size of the image, in GB? [64]:"
 read -r _finalsize
 
 if [[ -z $_finalsize ]]; then
-    _finalsize="64g"
+    _finalsize="64"
 fi
 
-$_compression -c -d $_path/"$_filename" | pv -s $_finalsize > /dev/nvme0n1
+_disk="/dev/nvme0n1"
+
+# Get our image size in raw bytes
+_size=$(numfmt --from=iec "${_finalsize}G")
+
+clear
+echo "Found the following disks to flash:"
+while true; do
+
+    # Get a list of all available disks large enough to take our image, sorted by size
+    IFS=$'\n' read -r -d '' -a disks <<< "$(lsblk -x SIZE -nblo NAME,SIZE,TYPE | grep "disk" | awk -v var="$_size" '$2 > var {print $1}')"
```

Comment:
What happens if this list is empty? Can we have some message about there not being a disk large enough?

Reply:
Good call, I forgot to add that. Will do.